### PR TITLE
Update Queue-Optimized.swift

### PR DESCRIPTION
**Pull request by Peeoner174**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### CODE REVIEW

**Comment by Peeoner174** [comment]
Added extension to safely retrieve array element

File: Queue/Queue-Optimized.swift [hunk 1]
```diff
@@ -45,3 +45,14 @@ public struct Queue<T> {
     }
   }
 }
+
+extension Array {
+
```
Comment (richard-ash) [inline Queue/Queue-Optimized.swift]: Let's remove this empty line

File: Queue/Queue-Optimized.swift [hunk 2]
```diff
@@ -45,3 +45,14 @@ public struct Queue<T> {
...
+extension Array {
+    
+    subscript(guarded idx: Int) -> Element? {
+        guard (startIndex..<endIndex).contains(idx) else {
+            return nil
+        }
+        return self[idx]
+    }
+
```
Comment (richard-ash) [inline Queue/Queue-Optimized.swift]: Same here

File: Queue/Queue-Optimized.swift (hunk 1 above)
Comment (Peeoner174) [reply Queue/Queue-Optimized.swift]: Thanks.

**Comment by Peeoner174** [comment]
> Cool update! A couple small nits then you're gtg 🙂
Thanks.

**Pull request merged**